components/input.d.ts

```diff
@@ -128,50 +128,53 @@ declare module '@salesforce/design-system-react/components/input' {
 		/**
 		 * Triggered when focus is removed.
 		 */
-		onBlur?: (v: any) => any;
+		onBlur?: React.FocusEventHandler<HTMLInputElement>;
 		/**
 		 * This callback fires when the input changes. Passes in `event, { value }`.
 		 */
 		onChange?: (
-			e: React.KeyboardEvent<HTMLInputElement>,
-			value: { value: any }
+			e:
+				| React.ChangeEvent<HTMLInputElement>
+				| React.KeyboardEvent<HTMLButtonElement>
+				| React.MouseEvent<HTMLButtonElement>,
```

Comment:
`ChangeEvent` is typical. I think that `KeyboardEvent` and `MouseEvent` will be encountered when `performStep` is invoked.